web/src/components/post-onboarding/PostOnboardingRadioQuestion.test.tsx

```diff
@@ -0,0 +1,161 @@
+            onboardingKey={onboardingKey}
+            taskId={taskId}
+          />
+        </WithStatefulUserData>,
+        IsAuthenticated.TRUE
+      )
+    );
+  };
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    setupStatefulUserDataContext();
+  });
+
+  it("sets given profileKey based on radio button toggle", async () => {
+    const initialBusiness = generateBusiness({
+      profileData: generateProfileData({ constructionRenovationPlan: undefined }),
+    });
+
+    mockFetchPostOnboarding.mockResolvedValue(generatePostOnboarding({}));
+    renderComponent(initialBusiness, "constructionRenovationPlan");
+
+    await waitFor(() => {
+      expect(screen.getByTestId(postOnboardingQuestionId)).toBeInTheDocument();
+    });
+
+    expect(screen.queryByTestId("post-onboarding-false-content")).not.toBeInTheDocument();
+    expect(screen.queryByTestId("post-onboarding-true-content")).not.toBeInTheDocument();
+
+    fireEvent.click(screen.getByTestId("post-onboarding-radio-true"));
+    expect(screen.queryByTestId("post-onboarding-false-content")).not.toBeInTheDocument();
+    expect(screen.getByTestId("post-onboarding-true-content")).toBeInTheDocument();
+    expect(currentBusiness().profileData.constructionRenovationPlan).toBe(true);
+
+    fireEvent.click(screen.getByTestId("post-onboarding-radio-false"));
+    expect(screen.getByTestId("post-onboarding-false-content")).toBeInTheDocument();
+    expect(screen.queryByTestId("post-onboarding-true-content")).not.toBeInTheDocument();
+
+    expect(currentBusiness().profileData.constructionRenovationPlan).toBe(false);
+  });
+
+  describe("checkboxes and task status updates", () => {
+    const setupAndRenderWithCheckboxes = async (): Promise<void> => {
+      const initialBusiness = generateBusiness({
+        profileData: generateProfileData({ constructionRenovationPlan: true }),
+        taskItemChecklist: {},
+        taskProgress: {
+          [taskId]: "NOT_STARTED",
+        },
+      });
+
+      mockFetchPostOnboarding.mockResolvedValue(
+        generatePostOnboarding({
+          contentMd: "- []{checkbox-1} content1\n" + "- []{checkbox-2} content2\n",
+        })
+      );
+
+      renderComponent(initialBusiness, "constructionRenovationPlan");
+      await waitFor(() => {
+        expect(screen.getByTestId(postOnboardingQuestionId)).toBeInTheDocument();
+      });
+    };
+
+    it("updates task progress to In-Progress if radio is Yes and one or more checkboxes completed", async () => {
+      await setupAndRenderWithCheckboxes();
+      fireEvent.click(getInputByName("content1"));
+      expect(getInputByName("content1").checked).toBe(true);
+      expect(currentBusiness().taskItemChecklist["checkbox-1"]).toEqual(true);
+      expect(currentBusiness().taskProgress[taskId]).toEqual("IN_PROGRESS");
+    });
+
+    it("updates task progress to Completed if radio is Yes and all checkboxes completed", async () => {
```

Comment:
I noticed the test makes reference to ALL checkboxes, but we're only testing for two. Should we be adding input for two add'l checkboxes since there are a total of four?

Reply:
It *is* testing all of them. There are only 2 in the context of this test. The entire test is never making use of the "real" markdown. It is testing against a mocked task with 2 checkboxes: `checkbox-1` and `checkbox-2`, which is set up on line 95 of this file.

Reply:
Yes, I saw the input was only two checkboxes and thought it may be okay b/c you were "feeding" the tests two checkboxes, but when I checked the UI, I noticed that they were four checkboxes. I do understand the need to be flexible, etc. Thanks, y'all!

Reply:
@AnJuHyppolite The test is using a mock, it is not testing the task that appears in the application UI